metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -197,24 +199,25 @@ class PackageProvider(
             )
             newImports ++ refsInImportsEdits
           case None =>
-            val refsInImportsEdits = importersAndParts.flatMap {
-              case (importer, importParts) =>
-                calcImportEditsForFileMove(
-                  importer,
-                  importParts,
-                  oldPkgParts,
-                  refsNames,
-                  newPkgName,
-                )
-            }
+            val (otherToImport, refsInImportsEdits) =
+              calcAllImportsEditsForFileMove(
+                importersAndParts,
+                oldPkgParts,
+                refs.map { case (name, sym) => (name.value, sym) }.toSet,
```

Comment:
Could you calculate this in `calcAllImportsEditsForFileMove` ? You would avoid having `Set[(String, String)]` parameter